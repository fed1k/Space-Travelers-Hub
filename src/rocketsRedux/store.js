import { legacy_createStore as createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { myReducer } from './fetchData';

const rocketstore = createStore(myReducer, applyMiddleware(thunk));

export default rocketstore;
