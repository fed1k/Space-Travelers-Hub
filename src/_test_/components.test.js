import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../rocketsRedux/store';
import Rocket from '../components/Rockets';
import Missions from '../Pages/missionPage';
import MyProfile from '../components/myProfile';

it('renders rockets in an expected way', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Rocket />
      {' '}

    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders missions in an expected way', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Missions />
      {' '}
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders profile in an expected way', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MyProfile />
      {' '}
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
