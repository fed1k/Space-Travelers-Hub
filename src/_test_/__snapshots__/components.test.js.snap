// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders missions in an expected way 1`] = `
Array [
  <section>
    <table>
      <thead>
        <tr>
          <th>
            Mission
          </th>
          <th>
            Description
          </th>
          <th>
            Status
          </th>
          <th>
            ChangeStatus
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </section>,
  " ",
]
`;

exports[`renders profile in an expected way 1`] = `
Array [
  <div
    className="profiles"
  >
    <div
      className="myMissions"
    >
      <div>
        <h1>
          My Missions
        </h1>
      </div>
    </div>
    <div
      className="myProfile"
    >
      <div
        className="rockets-container"
      >
        <h1>
          My Rockets
        </h1>
      </div>
    </div>
  </div>,
  " ",
]
`;

exports[`renders rockets in an expected way 1`] = `
Array [
  <section
    className="main-rocket-div"
  />,
  " ",
]
`;
